notebook/services/contents/filemanager.py

```diff
@@ -586,3 +590,23 @@ def get_kernel_path(self, path, model=None):
         else:
             parent_dir = ''
         return parent_dir
+
+    @staticmethod
+    def _validate_path(path):
+        """Checks if the path contains invalid characters relative to the current platform"""
+
+        if sys.platform == 'win32':
+            # On Windows systems, we MUST disallow colons otherwise an Alternative Data Stream will
+            # be created and confusion will reign! (See https://github.com/jupyter/notebook/issues/5190)
+            # Go ahead and add other invalid (and non-path-separator) characters here as well so there's
+            # consistent behavior - although all others will result in '[Errno 22]Invalid Argument' errors.
+            invalid_chars = (':', '>', '<', '*', '?', '|', '"')
+        else:
+            # On non-windows systems, allow the underlying file creation to perform enforcement when appropriate
+            invalid_chars = ()
+
+        for char in invalid_chars:
+            if char in path:
+                raise web.HTTPError(400, u"Path '{}' contains characters that are invalid for the filesystem. "
```

Comment:
The notebook is python 3 only now, right? So we don't need the `u` unicode identifier? `u"..."`

Reply:
Yes - good point.  Left other occurrences alone for now though.